pkg/access_log_server/service_v2.go

```diff
@@ -0,0 +1,139 @@
+package access_log_server
+
+import (
+	"io"
+	"strconv"
+
+	envoy_data_accesslog_v2 "github.com/envoyproxy/go-control-plane/envoy/data/accesslog/v2"
+	envoy_service_accesslog_v2 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/seznam/slo-exporter/pkg/event"
+	"github.com/seznam/slo-exporter/pkg/stringmap"
+)
+
+type AccessLogServiceV2 struct {
+	outChan chan *event.Raw
+	logger  logrus.FieldLogger
+	envoy_service_accesslog_v2.UnimplementedAccessLogServiceServer
+}
+
+func exportCommonPropertiesV2(p *envoy_data_accesslog_v2.AccessLogCommon) stringmap.StringMap {
+	res := stringmap.StringMap{}
+	res["DownstreamDirectRemoteAddress"] = p.DownstreamDirectRemoteAddress.String()
+	res["DownstreamLocalAddress"] = p.DownstreamLocalAddress.String()
+	res["DownstreamRemoteAddress"] = p.DownstreamRemoteAddress.String()
+	res["Metadata"] = p.Metadata.String()
+	res["ResponseFlags"] = p.ResponseFlags.String()
+	res["RouteName"] = p.RouteName
+	res["StartTime"] = p.StartTime.String()
+	res["TimeToFirstDownstreamTxByte"] = p.TimeToFirstDownstreamTxByte.String()
+	res["TimeToFirstUpstreamRxByte"] = p.TimeToFirstUpstreamRxByte.String()
+	res["TimeToFirstUpstreamTxByte"] = p.TimeToFirstUpstreamTxByte.String()
+	res["TimeToLastDownstreamTxByte"] = p.TimeToLastDownstreamTxByte.String()
+	res["TimeToLastRxByte"] = p.TimeToLastRxByte.String()
+	res["TimeToLastUpstreamRxByte"] = p.TimeToLastUpstreamRxByte.String()
+	res["TimeToLastUpstreamTxByte"] = p.TimeToLastUpstreamTxByte.String()
+	res["TlsProperties"] = p.TlsProperties.String()
+	res["UpstreamCluster"] = p.UpstreamCluster
+	res["UpstreamLocalAddress"] = p.UpstreamLocalAddress.String()
+	res["UpstreamRemoteAddress"] = p.UpstreamRemoteAddress.String()
+	res["UpstreamTransportFailureReason"] = p.UpstreamTransportFailureReason
+	res["SampleRate"] = strconv.FormatFloat(p.SampleRate, 'e', -1, 64)
+	// FilterStateObjects are omitted
+
+	return res
+}
+
+func exportLogEntriesV2(msg *envoy_service_accesslog_v2.StreamAccessLogsMessage) []stringmap.StringMap {
+	res := []stringmap.StringMap{}
+
+	if logs := msg.GetHttpLogs(); logs != nil {
+		for _, l := range logs.LogEntry {
+			exportedLogEntry := stringmap.StringMap{}
+			// AccessLogCommon
+			exportedLogEntry.Merge(exportCommonPropertiesV2(l.CommonProperties))
+
+			// ProtocolVersion
+			exportedLogEntry["ProtocolVersion"] = l.ProtocolVersion.String()
+
+			// Request
+			exportedLogEntry["Authority"] = l.Request.Authority
+			exportedLogEntry["ForwardedFor"] = l.Request.ForwardedFor
+			exportedLogEntry["OriginalPath"] = l.Request.OriginalPath
+			exportedLogEntry["Path"] = l.Request.Path
+			exportedLogEntry["Port"] = l.Request.Port.String()
+			exportedLogEntry["Referer"] = l.Request.Referer
+			exportedLogEntry["RequestHeaders"] = stringmap.StringMap(l.Request.RequestHeaders).String()
```

Comment:
My intention was to keep relation between envoy access_log protobufs 1:1 with the resulting metadata stringmap unless we have a very special reason to do otherwise. (https://github.com/seznam/slo-exporter/pull/37#discussion_r547243024 relates to that as well).

Reply:
our uses do not care about protobuf serialization (almost all do not know how the message is serialized), but what is important form them is easy and clean configuration of slo-exporter.